src/service_daemon.rs

```diff
@@ -1863,9 +1938,18 @@ impl Zeroconf {
 
         if !out.answers.is_empty() {
             out.id = msg.id;
-            broadcast_dns_on_intf(&out, intf_sock);
-
-            self.increase_counter(Counter::Respond, 1);
+            if is_unicast_reply {
+                if is_legacy_unicast {
+                    unicast_dns_on_intf(&out, src_addr.ip(), intf_sock);
+                    self.increase_counter(Counter::RespondLegacyUnicast, 1);
+                } else {
+                    unicast_dns_on_intf(&out, src_addr.ip(), intf_sock);
```

Comment:
nit pick:  this common call of `unicast_dns_on_intf` can be extracted out and moved to before `if is_legacy_unicast` check.